lib/source-code/token-store/utils.js

```diff
@@ -36,11 +30,10 @@ function getStartLocation(token) {
  * @returns {number} The found index or `tokens.length`.
  */
 exports.search = function search(tokens, location) {
-    return lodash.sortedIndexBy(
-        tokens,
-        { range: [location] },
-        getStartLocation
-    );
+    const val = getStartLocation({ range: [location] });
+    const index = tokens.findIndex(el => val <= getStartLocation(el));
+
+    return index === -1 ? tokens.length : index;
```

Comment:
Similar to the previous comment, `lodash.sortedIndexBy` most likely uses a binary search (and jsdoc description for this `function search` says "Binary-searches..."), while `findIndex` performs a linear search.

I think `utils.search` is used only on arrays of comments, which typically shouldn't be large arrays, so the difference _might_ be small.

@eslint/eslint-tsc thoughts about this, should we look for a binary search library?

Reply:
I also think the difference might be small.

By the way, `getStartLocation({ range: [location] })` is `{ range: [location] }.range[0]`, so it does nothing.

Reply:
Useless `val` declaration removed in 00ac18335.

Reply:
findIndex is slower, but not exponentially so, and both operations can run millions of times per second.

findIndex is slower, but not exponentially so, and both operations can run millions of times per second.

```
$ node benchmark-search.js
sortedIndexBy x 7,188,306 ops/sec ±0.84% (89 runs sampled)
Array#findIndex x 2,764,162 ops/sec ±1.65% (84 runs sampled)
```

Benchmark code: https://github.com/stephenwade/my-eslint-benchmark

Reply:
I used [hyperfine](https://github.com/sharkdp/hyperfine) to benchmark the test suite on this branch and the master branch. This branch ran slightly faster on my computer, although the results were very close.

<img width="584" alt="Screenshot of benchmark results running the npm test command. The remove-lodash-3 branch runs in about 172 seconds and the master branch runs in about 174 seconds." src="https://user-images.githubusercontent.com/4148577/118977433-24212a00-b944-11eb-9057-adf1813c54ab.png">

I believe that the speed difference in this method will not be noticeable and that we should stick with findIndex.